lib/active_storage_support/base64_attach.rb

```diff
@@ -5,13 +5,17 @@ module Base64Attach
 
     def attachment_from_data(attachment)
       if attachment.is_a?(Hash)
-        base64_data = attachment.delete(:data)
+        attachment = fill_attachment_data(attachment, attachment.delete(:data))
```

Comment:
I think you don't need to assign the value, you are already changing the attachment in the other method